include/private/tdymeshimpl.h

```diff
@@ -200,9 +213,15 @@ typedef struct TDyMesh {
   TDyRegion region_connected;
 
   PetscInt *closureSize, **closure, maxClosureSize;
+
+  PetscInt *nG2L; // Mapping of global cells to local cells
```

Comment:
I can't help but wonder if we should make use of PETSc's `IS`-related [types and functions](https://petsc.org/release/docs/manualpages/IS/index.html) for these mappings. This might help us anticipate and avoid issues with global-to-local mappings in large problems.,